sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt

```diff
@@ -258,24 +264,66 @@ internal class InternalPaymentViewModel(app: Application) : AndroidViewModel(app
         return getApplication<Application>().getString(resId)
     }
 
+
+
     sealed class UIState {
-        open fun getWebViewPage(vm: AndroidViewModel): String? = null
+        open val asHtmlContent: HtmlContent? get() = null
 
         object Loading : UIState()
-        class HtmlContent(@StringRes private val template: Int, private val link: String) : UIState() {
-            override fun getWebViewPage(vm: AndroidViewModel): String? {
-                return vm.getApplication<Application>().getString(template, link)
+        class HtmlContent(
+            val baseUrl: String?,
+            @StringRes private val template: Int,
+            private val link: String
+        ) : UIState() {
+            override val asHtmlContent get() = this
+            fun getWebViewPage(context: Context): String {
```

Comment:
Codacy found an issue: [UndocumentedPublicFunction - [getWebViewPage] at /src/sdk/src/main/java/com/swedbankpay/mobilesdk/internal/InternalPaymentViewModel.kt:279:17](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-android/pullRequest?prid=6212608)

Reply:
This is an internal function, not a public one